grafana-plugin/src/utils/hooks.tsx

```diff
@@ -138,3 +141,38 @@ export const useOnMount = (callback: () => void) => {
     callback();
   }, []);
 };
+
+export const useInitializePlugin = ({ meta }: AppRootProps) => {
+  const IS_OPEN_SOURCE = meta?.jsonData?.license === GRAFANA_LICENSE_OSS;
+  const [isInitialized, setIsInitialized] = useState(false);
+
+  // create oncall api token and save in plugin settings
+  const install = async () => {
+    await makeRequest(`/plugin${IS_OPEN_SOURCE ? '/self-hosted' : ''}/install`, {
+      method: 'POST',
+    });
+  };
+
+  const initializePlugin = async () => {
+    if (!meta?.secureJsonFields?.onCallApiToken) {
+      await install();
+    }
+
+    // trigger users sync
+    try {
+      await makeRequest(`/plugin/status`, {
+        method: 'POST',
+      });
+    } catch (_err) {
+      await install();
+    }
+
+    setIsInitialized(true);
```

Comment:
What if the install fails? What's the fallback now? Shouldn't we attempt to try it again at least?

Reply:
I don't think it makes sense to have second attempt, if it failed once, it would fail again. Anyway, this is kinda temporary solution because we hope to move this part to backend proxy too so that frontend doesn't call install explicitly